### DISCUSSION TITLE
Question on DNS

### DISCUSSION THREAD

**Discussion by ngx3** [chadgeary/cloudblock#75] [Q&A] [answered]
I'm looking to setup Cloudblock, but I'm wondering what it's using for a DNS recursive resolver? It's doesn't appear to be using DNSCrypt or Unbound. Or if it's not using anything at all?

All I see in the script is that it's possibly using Wireguard's DNS?

Is anyone using this setup that can verify?

**Comment by chadgeary** [comment, marked as answer]
Hey @ngx3 - cloudblock does not use a self-hosted recursive dns server. It uses public DNS over HTTPS endpoints (you choose one at installation time) via cloudflared.

This is a practical approach providing encryption between clients (laptop, cell phone, etc) all the way to a public DNS over HTTPS endpoint.

The DoH endpoints available include (from https://github.com/curl/curl/wiki/DNS-over-HTTPS):
`adguard applied-privacy cloudflare google hurricane-electric libre-dns opendns pi-dns quad9-recommended`

**Comment by neitsab** [comment]
Hi, setting up Cloudblock for the first time on OCI, I've been wondering what `pi-dns` corresponded to?
